src/index.js

```diff
@@ -63,13 +63,48 @@ module.exports = function (context, options) {
         }
       })
 
+      const versionRegex = options.versionRegex ? new RegExp(options.versionRegex, options.versionRegexOptions) : null;
+
       const addToSearchData = (d) => {
-        lunrBuilder.add({
+        // read the list of docusaurus versions
+        const docVersions = [
+          ...JSON.parse(
+            fs.readFileSync(
+              path.join(
+                path.resolve(context.siteDir, "docs"),
+                "..",
+                "versions.json"
+              ),
+              "utf-8"
+            )
+          ),
+          "next",
+        ];
+
+        // Determine the version of current document
+        let version = null;
```

Comment:
What happen in default docs, so without version?

Reply:
If versioning is disabled all docs will be searched.

Reply:
But the default version is the latest, and the common case. So if I search without select a version, I see al the versions in search?

Reply:
This part of the code assign to each document the version. The version is null when docs versioning is disabled.
The code which set the version to use in searching it's [src/theme/SearchBar/index.js](src/theme/SearchBar/index.js#L49-L57) .
If you search without select version you should get only results from the last version. Now it only happens when a user searches from the homepage.